docs/pages/Specs/nexus-prover.mdx

```diff
@@ -35,32 +35,124 @@ After receiving a Merkle proof from the initialization step with a Merkle proof
+
+### `bn254`-`grumpkin` curve cycle
+
+In order to be able to produce proofs that can be cheaply verified on Ethereum, the Nexus zkVM currently implements the Nova folding scheme using the pairing-friendly [`bn254`](https://neuromancer.sk/std/bn/bn254) as the primary curve, since the latter has precompile support on Ethereum. Moreover, since `bn254` forms a $2$-cycle along with the [`grumpkin`](https://hackmd.io/@aztec-network/ByzgNxBfd#2-Grumpkin---A-curve-on-top-of-BN-254-for-SNARK-efficient-group-operations) curve, `grumpkin` is used as the secondary curve.
+
+Interestingly, choosing `bn254` as the primary curve by itself is not enough to achieve cheap verification on Ethereum. For that, it is also important that later phases of the proof system, such as the compression phase, only require elliptic-curve operations on `bn254`. Unfortunately, this means that the IVC verifier for the secondary circuit will need to be implemented as a circuit over the `bn254` scalar field using non-native arithmetic.
+
+Hence, unlike Microsoft's reference implementation, the Nexus zkVM follows the approach proposed by CycleFold [[KS23](#references)] in its implementation of Nova to address concerns about non-native elliptic-curve arithmetic operations in later phases of the proof system. In this approach, instead of duplicating the IVC chain in both curves, the circuit implemented over the second curve is only used to perform a single group scalar multiplication over the base field of the first curve. Moreover, this is done in a way that allows the verifier in the first curve to avoid non-native arithmetic operations by verifiably delegating its group scalar multiplications to the second curve.
+
+Besides not having to reason about the 2-cycle of elliptic curves in the security proof of Nova [[KS23](#references)], one important advantage of CycleFold's approach over the one used in Microsoft's reference implementation is that the circuit size over the secondary curve becomes significantly smaller. More concretely, the size of the IVC verifier of the Nexus zkVM on the secondary curve is in the order of $3{,}000$ constraints.
+
+The reduction in circuit complexity over the secondary curve has, however, implications on the size of the IVC verifier over the primary curve since the latter circuit now needs to keep track of additional witnesses and to perform several additional small multi-scalar multiplications and hashing operations related to these witnesses. More concretely, the recursion overhead over the primary curve is in the order of $70{,}000$ constraints when used to realize an IVC scheme and $340{,}000$ constraints when used to realize a PCD scheme.
```

Comment:
Sequential Nova requires 120k constraints, 70k is achievable with emulated arithmetic optimization.

Reply:
I'm fixing the IVC overhead value in the text.